File: pset6/cash/cash.py
from numbers import Number
from cs50 import get_float
# gets input from user and ensures that only a positive number 
while True:
    change = get_float('Change Owed: ')
    change = round(change, 2) * 100
    if change > 0 and isinstance(change, Number):
        break
# stores the amount of coins used
coins = 0
# keeps track of quaters
while change >= 25:
    change -= 25
    coins += 1
# dimes
while change >= 10:
    change -= 10
    coins += 1
# nickels
while change >= 5:
    change -= 5
    coins += 1
# pennies    
while change >= 1:
    change -= 1
    coins += 1
# prints total number of coins used
print(f'{coins}'),

File: pset1/cash/cash.c
#include <stdio.h>
#include <math.h>

// defines the max number coin denominations
#define MAX_COINS 4

// stores coin denominations in coins_array
int coins_array[MAX_COINS] = {25, 10, 5, 1};

// prompt user for non zero positive value //
int get_number_coins(int change_remaining);

int main(void)
{
    // gets input on how much change is owed and stores it in variable
    float change;
    do
    {
        printf("change owed: ");
        scanf("%f", &change);
    }
    while (change <= 0);

    // multiply by 100 and round float value of change to $change
    int $change = (float) roundf(change * 100);

    // calls function that calculates the least number  of coins
    int coins_change = get_number_coins($change);

    // displays final amount of coins used to complete transaction
    printf("%d\n", coins_change);

}

// claculates the least number of coins needed to complete transaction
int get_number_coins(int change_remaining)
{

    int coins = 0;
    for (int i = 0; i < MAX_COINS; i++)
    {

        while (change_remaining >= coins_array[i])
        {
            coins++;
            change_remaining -= coins_array[i];
        }
        if (change_remaining == 0)
        {
            return coins;
        }

    }

    return coins;

}